### ISSUE TITLE
#28 PCGrad operates on per-parameter gradients instead of per-task gradients, corrupting MTL training

### ISSUE THREAD

**Issue by ionfwsrijan** [KanishJebaMathewM/Truxify#12340]
## Description
In `mtl/model.py`, PCGrad gradient surgery is applied over **per-parameter** gradients instead of **per-task** gradients:

```python
# mtl/model.py (~lines 256-264)
total_loss.backward()                       # sums ALL task losses into one p.grad per parameter
if self.gradient_method == 'pcgrad':
    grads = [p.grad for p in self.model.parameters() if p.grad is not None]  # per-PARAMETER
    processed_grads = self.gradient_surgery.pcgrad(grads)
    for p, g in zip([p for p in self.model.parameters() if p.grad is not None], processed_grads):
        p.grad = g

# GradientSurgery.pcgrad projects each gradient in the list against EVERY other:
#   for i: for j != i: if dot(grads[i], grads[j]) < 0: grads[i] -= projection
```

PCGrad is defined over **per-task** gradients (project task i's gradient to remove conflict with task j's). But `total_loss.backward()` has already summed all task losses, so `grads` is a list of hundreds of per-parameter aggregate gradients. The code therefore projects each parameter's gradient against every *other parameter's* gradient — mathematically meaningless cross-parameter "surgery."

## Expected Behavior
Compute separate `loss_t.backward(retain_graph=True)` per task, collect one gradient vector per task, then apply PCGrad across tasks.

## Actual Behavior
Gradients are perturbed across unrelated parameters, corrupting the multi-task optimization instead of resolving task conflicts.

## Proposed Fix
```python
task_grads = []
for t in range(num_tasks):
    self.model.zero_grad(); loss[t].backward(retain_graph=True)
    task_grads.append([p.grad.clone() for p in self.model.parameters() if p.grad is not None])
processed = self.gradient_surgery.pcgrad(task_grads)   # per-task
# write processed gradients back per parameter
```
Add a test asserting per-task vs per-parameter handling produces different, task-conflict-resolving updates.
